components/contract-functions/contract-function.tsx

```diff
@@ -149,25 +159,25 @@ export const ContractFunctionsPanel: React.FC<ContractFunctionsPanelProps> = ({
           pr={{ base: 0, md: 3 }}
           mb={{ base: 3, md: 0 }}
         >
-          {sortedFunctions.map((fn) => (
-            <ListItem key={fn.signature} my={0.5}>
+          {fnsOrEvents.map((fn) => (
+            <ListItem key={fn.name} my={0.5}>
               <Button
                 size="sm"
-                fontWeight={
-                  selectedFunction.signature === fn.signature ? 600 : 400
-                }
+                fontWeight={selectedFunction.name === fn.name ? 600 : 400}
                 leftIcon={
                   <Icon
                     boxSize={3}
                     as={
-                      fn.stateMutability === "view" ||
-                      fn.stateMutability === "pure"
-                        ? FiEye
-                        : FiEdit2
+                      isFunction
+                        ? (fn as AbiFunction).stateMutability === "view" ||
+                          (fn as AbiFunction).stateMutability === "pure"
+                          ? FiEye
+                          : FiEdit2
+                        : BsLightningCharge
```

Comment:
this is not gonna work, this will mean it's always an `FiEye` for events except for the selected event?

Reply:
I tested it and it works... not sure what you mean

Reply:
oh, the giant ternary statement confused me. hmmm